gravity-forms/gw-multiple-entries-list-field.php

```diff
@@ -76,32 +85,33 @@ public function create_multiple_entries( $entry ) {
 				$working_entry[ (string) $field_id ] = $this->_args['formatter']( $row[ $column - 1 ], $field_id, $this );
 			}
 
-			// by default, original entry is updated with list field data; if append_list_data is true,
 			if ( $index == 0 && ! $this->_args['append_list_data'] ) {
 
 				GFAPI::update_entry( $working_entry );
 
-				gform_add_meta( $working_entry['id'], 'gwmelf_parent_entry', true );
-				gform_add_meta( $working_entry['id'], 'gwmelf_group_entry_id', $working_entry['id'] );
-
 				/**
 				 * Sync the parent entry with our working entry so when it is passed onto other plugins using this filter,
 				 * it is up-to-date and if the entry is updated via this filter (looking at you, GFPaymentAddOn::entry_post_save()),
 				 * our changes will be preserved.
 				 */
-				$entry                          = $working_entry;
-				$entry['gwmelf_parent_entry']   = true;
-				$entry['gwmelf_group_entry_id'] = $working_entry['id'];
+				$entry = $working_entry;
```

Comment:
Does the removal of...

```
$entry['gwmelf_parent_entry']   = true;
$entry['gwmelf_group_entry_id'] = $working_entry['id'];
```

... still work with the `GFPaymentAddOn::entry_post_save()` scenario?

Reply:
Well, firstly, handling payments is a limitation - even before the changes in the PR - I'll soon add that to the snippet. E.g.: Just one list row is sent to Google Sheets for example, if the form is submitted with 2 list rows & needs payment, else if no payment is needed, the 2 rows are successfully added to GS.

Secondly, the code was not totally removed but moved out of the `foreach` since it should always run, and all it does is just to put some meta in the DB that identifies the 'parent' entry & also update the `$entry` object passed to the callback (since it's a filter) so that other callbacks that receive the entry afterwards & that might want to do something like `GFAPI::update_entry( $entry )` don't overwrite what (i.e. delete the new metadata) we just stored in the DB.

In fact, the former position of the code was preventing having the 'parent' entry properly marked (in the DB) in some cases - that was the issue David also noticed in [the video](https://www.loom.com/share/a4c9fb68ed0342908fcc58ede8e98ad5) (around 1:20).